src/condition.ts

```diff
@@ -19,7 +21,7 @@ export class Condition {
 		* @param target The root target entity the condition is associated with.
 		* @param properties The set of property paths specifying which properties and entities the condition should be attached to.
 		*/
-	constructor(type: ConditionType, message: string, target: Entity, properties: PropertyPath[] = []) {
+	constructor(type: ConditionType, message: string, target: Entity, source: Rule | FormatError, properties: PropertyPath[] = []) {
```

Comment:
The source argument isn't assigned in the constructor

Reply:
fixed.... that's an odd one. i think i started this before i got sick and completely forgot about it lol